components/support/crypto/nss_sys/src/lib.rs

```diff
@@ -0,0 +1,65 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#![allow(non_camel_case_types, non_upper_case_globals, non_snake_case)]
+
+mod types;
+
+use std::os::raw::{c_char, c_uchar};
+pub use types::nspr::*;
+pub use types::nss3::*;
+
+// This is the version this crate is claiming to be compatible with.
+// We check it at runtime using `NSS_VersionCheck`.
+pub const COMPATIBLE_NSS_VERSION: &str = "3.26";
+
+// Code adapted from https://stackoverflow.com/a/35591693. I am not this kind of smart.
```

Comment:
Is there a reason you can't just do?
```rust
    ($(unsafe fn $i:ident($($arg:ident: $argty:ty),*) -> $ret:ty;)*) => {$(
        #[cfg(not(target_os = "ios"))]
        lazy_static::lazy_static! {
            pub static ref $i: libloading::Symbol<'static, unsafe extern fn($($arg: $argty),*) -> $ret> = {
                unsafe {
                    LIBNSS3.get(stringify!($i).as_bytes()).expect(stringify!(Could not get $i handle))
                }
            };
        }
        #[cfg(target_os = "ios")]
        extern "C" {
            pub fn $i($($arg: $argty),*) -> $ret;
        }
    )*};

```

This will avoid us needing to expand the recursion limit if we add more than 128 functions.

Reply:
I assume that would mean that we only support functions that return in that case right? We might as well keep the flexibility (and I'm not quite sure we'll hit 128 functions anyway).

Reply:
You should be able to use `$(-> $ret:ty)?` for that, I think

Reply:
By the way this worked, thanks!